### DISCUSSION TITLE
Please support assignment operators within `with`expressions.

### DISCUSSION THREAD

**Discussion by otac0n** [dotnet/csharplang#7950] [Language Ideas]
For example:

    public record class State(double T, Camera Camera);

    ...

    return state with
    {
        T += elapsed.TotalSeconds,
        Camera ??= new(),
    };

I tried to use this and found it missing.

I expect this to be similar to the following:

    return state with
    {
        T = state.T + elapsed.TotalSeconds,
        Camera = state.Camera ?? new(),
    };

The difference I would expect is that the `state` expression would be evaluated only once with the former.

**Comment by 333fred** [comment]
Duplicate of https://github.com/dotnet/csharplang/issues/5176.

**Comment by otac0n** [comment]
That should arguably be two issues.  I don't think it makes as much sense in object initializers.
